src/InterNations/Sniffs/Syntax/MethodTypeHintsSniff.php

```diff
@@ -0,0 +1,324 @@
+        if ($file->findNext(T_WHITESPACE, $endBracket) === ($endBracket + 1)
+            && $tokens[$endBracket+2]['code'] === T_COLON) {
+            $error = sprintf(
+                'No space should come before colon, exactly one space after colon, expected return type formatting ',
+                $tokens[$returnTypeHintPtr]['content']
+            );
+            $error .= sprintf('to be "): %1$s" got ") : %1$s"', $tokens[$returnTypeHintPtr]['content']);
+            $file->addError($error, $namePtr, 'WrongStyleTypeHint');
+
+            return;
+        }
+
+        // PHP7 return type style check
+        if ($tokens[($endBracket + 1)]['code'] !== T_COLON) {
+            $error = sprintf(
+                'PHP 7 style return type hint, colon is required for method "%1$s::%2$s"',
+                $className,
+                $methodName
+            );
+            $file->addError($error, $namePtr, 'MissingReturnTypeHintColon');
+
+            return;
+        }
+
+        if (!in_array($tokens[$file->findNext(T_COLON, $endBracket)+2]['code'], [T_NULLABLE, T_RETURN_TYPE])) {
+            $error = 'Expected exactly one space after colon, multiple spaces or no space found for the return type...';
+            $file->addError($error, $namePtr, 'WrongStyleTypeHint');
+
+            return;
+        }
+
+        // Check for mandatory return type hints
+        if (isset(self::$whitelist[$methodName][1])
+            && self::$whitelist[$methodName][1] !== $tokens[$returnTypeHintPtr]['content']
+            && self::$whitelist[$methodName][1] !== false
+        ) {
+            $error = sprintf(
+                'Magic method "%1$s::%2$s" can only contain "%3$s" as a return type hint',
+                $className,
+                $methodName,
+                self::$whitelist[$methodName][1]
+            );
+            $file->addError($error, $returnTypeHintPtr, 'WrongReturnTypeHint');
+
+            return;
+        }
+
+        // Check for array return type hing
+        if ($tokens[$returnTypeHintPtr]['content'] === 'array' && empty($returnDoc)) {
+            $error = sprintf(
+                'Return type hint for a method "%1$s::%2$s" must be documented to specify their exact type, ',
+                $className,
+                $methodName
+            );
+            $error .= 'Use "@return Class[]" for a list of classes, use "@return integer[]" for a list of integers ';
+            $error .= 'and so on...';
+            $file->addError($error, $returnTypeHintPtr, 'MissingDocForReturnTypeHint');
+
+            return;
+        }
+
+
+        // Catch Superfluous return comment doc
+        if ($tokens[$returnTypeHintPtr]['content'] !== 'array' && $returnDoc) {
+            $error = 'Superfluous return type doc';
+            $file->addError($error, array_keys($returnDoc)[0], 'superfluousParamDoc');
+        }
+    }
+
+    private function ifParamDocExistForArrayParameter($needle, &$paramDoc)
```

Comment:
`unsetrParamType`?
Also: you $needle should probably be $paramName and $paramDoc needs an array type hint. Could also use a bool return type hint